### PR TITLE
Insights/context filter prototype

### PR DESCRIPTION
**Pull request by coury-clark**
Adds query filter arguments for search contexts to the `InsightView` resolver. Future PRs will add this to the edit schema, and add it to the just-in-time path, and the applied / default schema.

## Test plan

``` gql
query insightview{
  insightViews (filters:{searchContexts:["@dev/srcall"]}) {
    nodes {
      id
      appliedFilters {
        excludeRepoRegex
        includeRepoRegex
      }
      defaultFilters {
        excludeRepoRegex
        includeRepoRegex
      }
      dataSeriesDefinitions {
        ... on SearchInsightDataSeriesDefinition {
          query
        }
      }
      dataSeries {
        points {
          value
          dateTime
        }
      }
    }
  }
}
```

### Capture group series
``` gql
query insightview{
  insightViews (id:"aW5zaWdodF92aWV3OiIyN2hpcU51em9QVHVzZGxPWE9HRG9xS2xxUlgi", filters:{searchContexts:["@dev/hashicorp"]}) {
    nodes {
      id
      appliedFilters {
        excludeRepoRegex
        includeRepoRegex
      }
      defaultFilters {
        excludeRepoRegex
        includeRepoRegex
      }
      dataSeriesDefinitions {
        ... on SearchInsightDataSeriesDefinition {
          query
        }
      }
      dataSeries {
        label
        points {
          value
          dateTime
        }
      }
    }
  }
}
```

``` json
{
  "data": {
    "insightViews": {
      "nodes": [
        {
          "id": "aW5zaWdodF92aWV3OiIyN2hpcU51em9QVHVzZGxPWE9HRG9xS2xxUlgi",
          "appliedFilters": {
            "excludeRepoRegex": null,
            "includeRepoRegex": null
          },
          "defaultFilters": {
            "excludeRepoRegex": "",
            "includeRepoRegex": ""
          },
          "dataSeriesDefinitions": [
            {
              "query": "file:go\\.mod$ go\\s*(\\d\\.\\d+) patterntype:regexp"
            }
          ],
          "dataSeries": [
            {
              "label": "1.13",
              "points": [
                {
                  "value": 1,
                  "dateTime": "2021-05-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-06-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-07-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-08-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-09-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-10-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-11-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2021-12-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2022-01-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2022-02-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2022-03-12T00:00:00Z"
                },
                {
                  "value": 1,
                  "dateTime": "2022-04-12T00:00:00Z"
                }
              ]
            }
          ]
        }
      ]
    }
  }
}
```

### CODE REVIEW

File: cmd/frontend/graphqlbackend/insights.graphql [hunk 1]
```diff
@@ -622,6 +622,8 @@ input InsightViewFiltersInput {
     A regex string for which to exclude repositories in a filter.
     """
     excludeRepoRegex: String
+
+    searchContexts: [String!]
```
Comment (coury-clark) [inline cmd/frontend/graphqlbackend/insights.graphql:626]:
```suggestion
    """
    A list of query based search contexts to include in the filters for the view.
    """
    searchContexts: [String!]
```

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go [hunk 2]
```diff
@@ -175,11 +175,18 @@ func expandCaptureGroupSeriesRecorded(ctx context.Context, definition types.Insi
...
+		opts.ExcludeRepoRegex = append(opts.ExcludeRepoRegex, regex...)
+	}
+
 	if filters.IncludeRepoRegex != nil {
-		opts.IncludeRepoRegex = *filters.IncludeRepoRegex
+		includeRepo(*filters.IncludeRepoRegex)
 	}
 	if filters.ExcludeRepoRegex != nil {
-		opts.ExcludeRepoRegex = *filters.ExcludeRepoRegex
+		excludeRepo(*filters.ExcludeRepoRegex)
```
[the remark this answers is not in the document]
Comment (coury-clark) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:189]: Fixed on latest and I added an example filtered response to the OP

[the remark this answers is not in the document]
Comment (chwarwick) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:189]: Agree I took me reading though a few times to figure out if this was handled somewhere else and understanding what was different between resolving a `view` vs a `series`, which I'm still not totally clear on, I need to spend some time looking at the schema and pair it up with resolvers.

**Review by chwarwick — APPROVED** [review]
LGTM, did see some extra logging of the [include/exclude preds](https://github.com/sourcegraph/sourcegraph/pull/33686/files#diff-a98fbf642c72d8ea4a9bf1344828af25a3fe2eb751c7b483d6e12d2f0a5f465bR265) that I thought might be debugging related.

File: enterprise/internal/insights/resolvers/insight_view_resolvers.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (coury-clark) [reply enterprise/internal/insights/resolvers/insight_view_resolvers.go:189]: Here is the issue I've opened https://github.com/sourcegraph/sourcegraph/issues/33804, it has a notebook with some more context

File: enterprise/internal/insights/store/store.go [hunk 3]
```diff
@@ -246,10 +248,22 @@ func seriesPointsQuery(opts SeriesPointsOpts) *sqlf.Query {
 		preds = append(preds, sqlf.Sprintf(s))
 	}
 	if len(opts.IncludeRepoRegex) > 0 {
-		preds = append(preds, sqlf.Sprintf("rn.name ~ %s", opts.IncludeRepoRegex))
+		for _, regex := range opts.IncludeRepoRegex {
+			if len(regex) == 0 {
+				continue
+			}
+			preds = append(preds, sqlf.Sprintf("rn.name ~ %s", regex))
+			log15.Info("include", "preds", preds)
```
Comment (coury-clark) [inline enterprise/internal/insights/store/store.go:256]:
```suggestion
```

**Comment by coury-clark** [comment]
> LGTM, did see some extra logging of the [include/exclude preds](https://github.com/sourcegraph/sourcegraph/pull/33686/files#diff-a98fbf642c72d8ea4a9bf1344828af25a3fe2eb751c7b483d6e12d2f0a5f465bR265) that I thought might be debugging related.

I'm nothing if not messy 😆 

Fixed in the latest, thanks!

**Pull request merged**